### DISCUSSION TITLE
Simulation fails because of getting negative concentrations

### DISCUSSION THREAD

**Discussion by mmalsmadi** [Open-Systems-Pharmacology/Forum#849] [Questions & Problems]
Hello 

I had this error when running the PK-Sim
![image](https://user-images.githubusercontent.com/37537530/98109632-89c92f80-1ea6-11eb-97f2-9b1ae0084450.png)

I exported the simulation to MoBi and I found that some parameters missing their values, so I filled them, and ran the simulation again and I got the same PK-Sim error message
![image](https://user-images.githubusercontent.com/37537530/98109715-b0876600-1ea6-11eb-9187-cd3e0d5d8111.png)

I check the ceacum parameters in MoBi and found the following being negative for the estimation of thickness, is that OK?
![image](https://user-images.githubusercontent.com/37537530/98109479-51295600-1ea6-11eb-8669-9ad0a7c1f267.png)

how to fix the issue?

Thanks

**Comment by StephanSchaller** [comment]
> and I found that some parameters missing their values, so I filled them

Can you elaborate what you mean by this?

**Comment by AndreDlm** [comment]
Hi @mmalsmadi,
did you try reducing the solver tolerances?
Negative values for the parameters used to calculate thickness are nothing to worry about; in any case, the calculated thickness should be a positive value:
![image](https://user-images.githubusercontent.com/26541603/98140066-429e6700-1ec5-11eb-9ca4-34360a9d6ef2.png)

**Comment by mmalsmadi** [comment]
please check this from the MoBi simulation exported from PK-Sim

![NA values in MoBi from PKSim](https://user-images.githubusercontent.com/37537530/98270528-a1ccac00-1f97-11eb-935d-017865d7a6e2.png)


Regards,

Mo'tasem Alsmadi, PhD
Graduate Studies Assistant Dean
Assistant Professor
Department of Pharmaceutical Technology
Faculty of Pharmacy
Jordan University of Science & Technology
P.O.Box 3030 - Irbid 22110, Jordan
Phone: +96227201000    Ext. 23522,  26168
________________________________

**Comment by mmalsmadi** [comment]
I did as shown here from Pk-Sim simulation, and I got the same error messege


![absolute and relative tolerances](https://user-images.githubusercontent.com/37537530/98270626-bad55d00-1f97-11eb-803c-233b77dceb41.png)


Regards,

Mo'tasem Alsmadi, PhD
Graduate Studies Assistant Dean
Assistant Professor
Department of Pharmaceutical Technology
Faculty of Pharmacy
Jordan University of Science & Technology
P.O.Box 3030 - Irbid 22110, Jordan
Phone: +96227201000    Ext. 23522,  26168
________________________________

**Comment by AndreDlm** [comment]
@mmalsmadi did you sent any images in your last comments? Images are removed when answering via email, so you may want to upload them again

**Comment by mmalsmadi** [comment]
please check this from the MoBi simulation exported from PK-Sim where NA values appear
![NA values in MoBi from PKSim](https://user-images.githubusercontent.com/37537530/98270870-fe2fcb80-1f97-11eb-9a6d-c169ade4208f.png)


I decreased the absolute and relative tolerances, but I got the same error message as before

![absolute and relative tolerances](https://user-images.githubusercontent.com/37537530/98270987-21f31180-1f98-11eb-8882-9793fffdd010.png)

**Comment by msevestre** [comment]
@mmalsmadi THis is for the undefined liver enzyme. This is fine. NA are for properties such as Molecular Weight, Lipo etc that are typically user inputs. In the case of an enzyme there are not required

**Comment by StephanSchaller** [comment]
And the "NaN" are sometimes popping up where there is a formula behind the value and those will then onlz be shown as numbers in the simulation.

The solver tolarances need to be changed in the simulation, not in the utilities.

**Comment by mmalsmadi** [comment]
does this apply to "undefined liver metabolites"
If so, what other things could cause this problem?

Regards,

Mo'tasem Alsmadi, PhD
Graduate Studies Assistant Dean
Assistant Professor
Department of Pharmaceutical Technology
Faculty of Pharmacy
Jordan University of Science & Technology
P.O.Box 3030 - Irbid 22110, Jordan
Phone: +96227201000    Ext. 23522,  26168
________________________________

**Comment by mmalsmadi** [comment]
I got the following error message

![image](https://user-images.githubusercontent.com/37537530/98284341-a1d5a780-1fa9-11eb-8ddb-fef0b37f4d95.png)


after changing the settings of the simulation

![image](https://user-images.githubusercontent.com/37537530/98284419-c3cf2a00-1fa9-11eb-9b6c-ca438283987b.png)

**Comment by StephanSchaller** [comment]
it can be caused by processes you define if they are not well-defined... likely, this happens if sth with a very low concentration interacts with sth with a very high concentration

try going to 1e-15 with absolute tolerance. If this does not solve the issue, your model is likely not well defined and I would suggest to re-create it and revisit/check the parameter values you put in.

**Comment by Yuri05** [comment]
@mmalsmadi Can you upload your original **PK-Sim** simulation?

**Comment by mmalsmadi** [comment]
when I changed the setting to be non-Jackobean, it worked, what could be the problem in this case?

![image](https://user-images.githubusercontent.com/37537530/98363650-fa538600-2037-11eb-847a-19b450cb379f.png)

**Comment by AndreDlm** [comment]
Without the original PK-Sim simulation, it's hard to tell what went wrong. Using the Jacobian matrix can speed up the solving of your ODE system. See here for more info on the solver settings in PK-Sim:
https://docs.open-systems-pharmacology.org/working-with-mobi/mobi-documentation/setting-up-simulation#ode-solver-properties
